File: Week 2/A/deforestation.cpp
#include <bits/stdc++.h>
using namespace std;
int orig[1000005], PSA[1000005], N, Q, a, b; 
int main() {
    cin >> N; 
    for(int i = 1; i <= N; i++) {
        cin >> orig[i]; 
    }
    for(int i = 1; i <= N; i++) {
        PSA[i] = PSA[i-1] + orig[i]; 
    }
    cin >> Q; 
    for(int i = 1; i <= Q; i++) {
        cin >> a >> b;
        a++; // add one since queries are 0-indexed 
        b++; 
        cout << PSA[b] - PSA[a-1] << "\n"; 
    }
}

File: Week 2/A/deforestation.py
n=int(input())
orig=[0] # 1-indexed, add element at start of array
for i in range(n):
    orig.append(int(input()))
PSA=[0]
for i in range(1, n+1):
    PSA.append(PSA[i-1] + orig[i]) 
for i in range(int(input())): # loop through each query
    a,b=map(int,input().split())
    a+=1 # add one since queries are 0-indexed 
    b+=1 # but PSA should be 1 indexed
    print(PSA[b]-PSA[a-1])
